CHANGELOG.md

```diff
@@ -6,7 +6,147 @@ document.
+* [`let_and_return`] fixed false positive involving short-lived block temporary variables [#14180](https://github.com/rust-lang/rust-clippy/pull/14180)
+* [`manual_async_fn`] no longer emits suggestions inside macros [#14142](https://github.com/rust-lang/rust-clippy/pull/14142)
+* [`use_self`] skips analysis inside macro expansions of a `impl Self` block [#13128](https://github.com/rust-lang/rust-clippy/pull/13128)
+* [`manual_ok_err`] fixed replacement expression when it follows an `else` [#14240](https://github.com/rust-lang/rust-clippy/pull/14240)
+* [`double_ended_iterator_last`] no longer triggers on non-reference immutable receiver [#14140](https://github.com/rust-lang/rust-clippy/pull/14140)
+
+### Suggestion Fixes/Improvements
+
+* [`from_over_into`] fixed invalid code suggestion when self parameter has type annotation [#14409](https://github.com/rust-lang/rust-clippy/pull/14409)
+* [`needless_return`] now makes correct suggestions which require enclosing parentheses [#14094](https://github.com/rust-lang/rust-clippy/pull/14094)
+* [`match_bool`] fixed suggestion when the rewritten block has a label [#14102](https://github.com/rust-lang/rust-clippy/pull/14102)
+* [`obfuscated_if_else`] fixed suggestion by issuing required parentheses around the left side of a binary expression [#14124](https://github.com/rust-lang/rust-clippy/pull/14124)
+* [`nonminimal_bool`] added required parentheses when negating a binary expression [#14187](https://github.com/rust-lang/rust-clippy/pull/14187)
+
+### ICE Fixes
+
+* [`macro_use_imports`] Fix ICE when checking attributes
+  [#14317](https://github.com/rust-lang/rust-clippy/pull/14317)
+* [`doc_nested_refdefs`] Fix ICE by avoiding invalid ranges
+  [#14308](https://github.com/rust-lang/rust-clippy/pull/14308)
+* [`just_underscores_and_digits`] Fix ICE in error recovery scenario
+  [#14168](https://github.com/rust-lang/rust-clippy/pull/14168)
+* [`declare_interior_mutable_const`], [`borrow_interior_mutable_const`] Fix ICE by properly resolving `<T as Trait>::AssocT` projections
+  [#14125](https://github.com/rust-lang/rust-clippy/pull/14125)
+
+### Documentation Improvements
+
+* [`struct_excessive_bools`] Documentation improved with rationale
+  [#14351](https://github.com/rust-lang/rust-clippy/pull/14351)
+
+### Others
+
+* Use edition=2021 in `rustc_tools_util`
+  [#14211](https://github.com/rust-lang/rust-clippy/pull/14211)
+* Fix rustc_tools_util's `version.host_compiler` release channel, expose the rustc version, and add tests
+  [#14123](https://github.com/rust-lang/rust-clippy/pull/14123)
+* Make UI test annotations mandatory
+  [#11421](https://github.com/rust-lang/rust-clippy/pull/11421)
+* Add missing tests annotations for `ui-internal`
+  [#14388](https://github.com/rust-lang/rust-clippy/pull/14388)
+* Make internals ui tests annotations mandatory
+  [#14393](https://github.com/rust-lang/rust-clippy/pull/14393)
```

Comment:
This seems purely internal, not sure if we included these kinds of internal-only changes in the changelog before since it's a more user-facing document. I don't mind it either way, since it is a larger internal change (I feel like these three could be merged into one point as just "Make UI tests mandatory" though)